js/state/frameStateUtil.js

```diff
@@ -717,6 +722,38 @@ const setTabHoverState = (state, frameKey, hoverState, enablePreviewMode) => {
   return state
 }
 
+const getTabPageIndex = (state) => {
+  const tabPageIndex = state.getIn(['ui', 'tabs', 'tabPageIndex'], 0)
+  const previewTabPageIndex = state.getIn(['ui', 'tabs', 'previewTabPageIndex'])
+
+  return previewTabPageIndex != null ? previewTabPageIndex : tabPageIndex
+}
+
+const getCurrentTabPageFrameKeys = (state) => {
+  const pageIndex = getTabPageIndex(state)
+  const tabsPerTabPage = Number(getSetting(settings.TABS_PER_PAGE))
+  const startingFrameIndex = pageIndex * tabsPerTabPage
+  const unpinnedTabs = getNonPinnedFrames(state) || Immutable.List()
+
+  return unpinnedTabs
+    .slice(startingFrameIndex, startingFrameIndex + tabsPerTabPage)
+    .map((tab) => tab.get('key'))
+}
+
+const isFirstFrameKeyInTabPage = (state, frameKey) => {
+  return getCurrentTabPageFrameKeys(state).first() === frameKey
+}
+
+const hasFrame = (state, frameKey) => {
+  const frame = getFrameByKey(state, frameKey)
+  return !frame.isEmpty()
```

Comment:
`getFrameByKey` can return null as well so we need to add check here as well `frame && !frame.isEmpty()`